templates/boltzmann/core/honeycomb.ts

```diff
@@ -0,0 +1,909 @@
+  // otherwise sets up the beeline library.
+  // This needs to occur before any imports you want instrumentation
+  // to be aware of.
+  public init(): void {
+    if (!this.features.honeycomb) return
+
+    const writeKey = this.getWriteKey()
+    const dataset = this.getDataset()
+    const sampleRate = this.options.sampleRate || 1
+    const serviceName = this.options.serviceName
+
+    if (!this.features.otel) {
+      beeline({ writeKey, dataset, sampleRate, serviceName })
+      return
+    }
+
+    const f = this.factories
+    const apiHost: string = this.getApiHost()
+
+    const metadata: Metadata = f.metadata(writeKey, dataset)
+
+    const sampler: Sampler = f.sampler(sampleRate)
+    const exporter = f.traceExporter(apiHost, metadata)
+    const processor = f.spanProcessor(exporter)
+    const instrumentations = f.instrumentations()
+
+    const provider: NodeTracerProvider = f.tracerProvider(sampler)
+    provider.addSpanProcessor(processor)
+    provider.register()
+
+    this.sdk = f.sdk(
+      serviceName,
+      instrumentations,
+      exporter
+    )
+
+    this.initialized = true
+  }
+
+  // Start the OpenTelemetry SDK. If using beelines, this is
+  // a no-op. This needs to happen before anything happens in
+  // the entrypoint and is an async operation.
+  public async start(): Promise<void> {
+    let exitCode = 0
+    const sdk = this.sdk
+
+    async function die(err: Error) {
+      console.error(err.stack)
+      exitCode = 1
+      await shutdown()
+    }
+
+    async function shutdown() {
+      if (sdk) {
+        try {
+          await sdk.shutdown()
+        } catch (err) {
+          console.error(err.stack)
+        }
+      }
+      process.exit(exitCode)
+    }
+
+    if (sdk) {
+      process.once('SIGTERM', shutdown)
+      process.once('beforeExit', shutdown)
+      process.once('uncaughtException', die)
+      process.once('unhandledRejection', die)
+      await sdk.start()
+      this.started = true
+    }
+  }
+
+  // Create a trace, call the runInContext function, then close the
+  // trace after it resolved. For beeline, this is is a Trace;
+  // for OpenTelemetry, it's a root span.
+  public async withTrace(
+    context: Context,
+    runInContext: () => Promise<any>,
+    headerSources?: string[]
+  ): Promise<any> {
+    if (!this.features.honeycomb) {
+      // Call legacy beelines implementation
+      return this.withBeelineTrace(context, runInContext, headerSources)
+    }
+
+    if (headerSources) {
+      console.warn('trace headerSources are a beeline-only feature')
+    }
+
+    /*
+     * ┏┓
+     * ┃┃╱╲ in
+     * ┃╱╱╲╲ this
+     * ╱╱╭╮╲╲house
+     * ▔▏┗┛▕▔ we
+     * ╱▔▔▔▔▔▔▔▔▔▔╲
+     * trace with opentelemetry
+     * ╱╱┏┳┓╭╮┏┳┓ ╲╲
+     * ▔▏┗┻┛┃┃┗┻┛▕▔
+     */
+
+    const tracer = this.tracer
+    let carrier = {}
+
+    // Start a parent span
+    const parentSpan = tracer.startSpan(
+      `${context.method} ${context.url.pathname}${context.url.search}`,
```

Comment:
I was talking with ceej, and we're pretty sure that we want to remove `${context.url.search}` here (and also in the same line of the OTel code)
we don't want the query params as part of the span name, that will cause issues.

We should probably take a look at the span names that are being created once you have the test stuff up and running, and then take care of this.